homeassistant/components/automate/sensor.py

```diff
@@ -0,0 +1,101 @@
+"""Support for Automate Roller Blind Batteries."""
+from homeassistant.const import (
+    ATTR_VOLTAGE,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .base import AutomateBase
+from .const import AUTOMATE_HUB_UPDATE, DOMAIN
+from .helpers import async_add_automate_entities
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Automate Rollers from a config entry."""
+    hub = hass.data[DOMAIN][config_entry.entry_id]
+
+    current_battery = set()
+    current_signal = set()
+
+    @callback
+    def async_add_automate_sensors():
+        async_add_automate_entities(
+            hass, AutomateBattery, config_entry, current_battery, async_add_entities
+        )
+        async_add_automate_entities(
+            hass, AutomateSignal, config_entry, current_signal, async_add_entities
+        )
+
+    hub.cleanup_callbacks.append(
+        async_dispatcher_connect(
+            hass,
+            AUTOMATE_HUB_UPDATE.format(config_entry.entry_id),
+            async_add_automate_sensors,
+        )
+    )
+
+
+class AutomateBattery(AutomateBase):
+    """Representation of a Automate cover battery sensor."""
+
+    device_class = DEVICE_CLASS_BATTERY
```

Comment:
Please use the official class attributes instead:
https://developers.home-assistant.io/docs/core/entity#entity-class-or-instance-attributes

Reply:
To clarify, you mean use something like the following?
```
_attr_device_class = DEVICE_CLASS_BATTERY
```
(It's a bit confusing as the property method does not do this.)

Reply:
Yes, exactly. Here's the implementation:
https://github.com/home-assistant/core/blob/10bfc783651a49ffbe2e7ad8d3e5d5e3223b3180/homeassistant/helpers/entity.py#L334-L341